### ISSUE TITLE
Support python 3.12

### ISSUE THREAD

**Issue by SimonMaenaut** [wdpozzo/raynest#3]
These are the current blockers I am aware of preventing raynest from supporting python 3.12 at the moment.
- [x] Ray does not support python 3.12 yet (see https://github.com/ray-project/ray/issues/40211 ).

**Comment by GCArullo** [comment]
Looks like this is solved now?

**Comment by SimonMaenaut** [comment]
Yes, since the end of October for python 3.12 and a few days for python 3.13 https://github.com/ray-project/ray/issues/49738 .

However, `setuptools` and `psutil` have to be added as dependecies.

**Comment by GCArullo** [comment]
@wdpozzo I address this [here](https://github.com/wdpozzo/raynest/pull/5), could you please take a look and merge if it works for you?
